ocfweb/docs/docs/services/web/django.md

```diff
@@ -1,91 +1,97 @@
+   In the steps below, we'll assume you put it at `~/app` in your home
+   directory.
 
-         tsunami$ mkdir ~/projects
+2. Inside your project directory, create a virtualenv and install your
+   dependencies. For example, to create a Python 3 virtualenv, try:
 
-1.   Go into the directory you just created:
+   ```bash
+   user@tsunami:~/app$ virtualenv -ppython3 venv
+   user@tsunami:~/app$ venv/bin/pip install django
+   ```
 
-         tsunami$ cd ~/projects
+   You'll want to install at least `django` to start with. If you're using
+   other packages, install those too. You can also install from a
+   `requirements.txt` file, if you have one.
 
-1.   Create a new Django project. For this example, the project name will be
-     "hatsune". You will want to stick with Python-friendly package and module
-     names, as detailed in [PEP
-     8](https://www.python.org/dev/peps/pep-0008/#package-and-module-names).
+   For full details on how to use pip, see [the pip documentation][pip-docs].
 
-         tsunami$ django-admin startproject hatsune
+3. Inside that same virtualenv, install the `flup6` package. You can just
+   copy-and-paste the below snippet:
 
-1.   Go into the project directory you just created and edit the settings.py
-     file with the necessary information:
+   ```bash
+   user@tsunami:~/app$ venv/bin/pip install flup6
+   ```
 
-         tsunami$ cd hatsune
-         tsunami$ vim settings.py
+   If you're using a `requirements.txt` file, you might want to add it there as
+   well. This part varies by how you're organizing your project.
 
-1.   Change the permissions of the settings.py file, since it might have
-     sensitive information like your database password!
+4. Make a directory under `public_html` to house your application. For example,
+   `~/public_html/django`. You can also just use `public_html` directly if
+   desired.
 
-         tsunami$ chmod go-rwx settings.py
+5. In the directory you just created, make a file called `run.fcgi` with the file contents:
 
-1.   Create an app, edit it, and so on:
+   ```python
+   #!/home/u/us/user/app/venv/bin/python
+   import os
+   import sys
 
-         tsunami$ django-admin startapp miku
-         tsunami$ vim miku/views.py
-         ...
-         tsunami$ vim urls.py
+   sys.path.insert(0, os.path.expanduser('~/app'))
 
-1.    Make a directory in your ~/public_html directory:
+   from flup.server.fcgi import WSGIServer
+   from djangoapp import wsgi
 
-        tsunami$ mkdir ~/public_html/hatsune
+   if __name__ == '__main__':
+       WSGIServer(wsgi.application).run()
+   ```
 
-1.    Go into the directory you just created, and create two files: *.htaccess*
-      (note the dot), and *run.fcgi*.
+   Make sure to replace the first line of `run.fcgi` file with the actual path
+   to your project's virtualenv Python. You can find it by running the command
+   `readlink -f ~/app/venv/bin/python`.
```

Comment:
i included that because you can't put `~/apps/...` in a shebang (no squiggly expansion)

i figured it was easier to say "run `readlink -f ~/apps/...`" than "use the full path, such as /home/c/ck/ckuehl" and explain to use the first letter, then the first two letters, than the full name